src/main/java/at/ac/tuwien/kr/alpha/solver/heuristics/BerkMin.java

```diff
@@ -189,11 +193,24 @@ private void pushToStack(NoGood noGood) {
 	}
 
 	private void incrementActivityCounter(int literal) {
-		activityCounters.compute(atomOf(literal), (k, v) -> (v == null ? DEFAULT_ACTIVITY : v) + 1);
+		int atom = atomOf(literal);
+		if (isAtomChoicePoint.test(atom)) {
+			activityCounters.compute(atom, (k, v) -> (v == null ? DEFAULT_ACTIVITY : v) + 1);
+		}
+		// TODO: check performance
+		// note that here (and in incrementSignCounter) we only count atoms that are
+		// choice points, which might affect performance.
+		// alternative approaches:
+		// 1. count everything and from time to time do a garbage collection (i.e.
+		// remove atoms that are no choice points)
+		// 2. make check cheaper, e.g. by using dedicated atom IDs (e.g. even
+		// integers for rule bodies, uneven for other atoms)
 	}
 	
 	private void incrementSignCounter(Integer literal) {
-		signCounters.compute(atomOf(literal), (k, v) -> (v == null ? DEFAULT_SIGN_COUNTER : v) + 1);
+		if (isAtomChoicePoint.test(atomOf(literal))) {
+			signCounters.compute(literal, (k, v) -> (v == null ? DEFAULT_SIGN_COUNTER : v) + 1);
```

Comment:
No, this is a correct bugfix, see d9bf7b6715a6e5173696b65661e0794b60f332f0.
In `incrementActivityCounter`, we are counting the activity of _atoms_, no matter whether they are positive or negative. In `incrementSignCounter`, however, we count how often an atom occurs positively in nogoods and how often it occurs negatively, therefore we associate the counters with _literals_.
But thank you for reviewing so attentively :)